src/System.Linq/src/System/Linq/Union.cs

```diff
@@ -102,21 +160,9 @@ public override bool MoveNext()
                 return false;
             }
 
-            public override void Dispose()
+            internal override Set<TSource> FillSet()
```

Comment:
Same here... why not just have this in the base implemented in terms of GetSourceEnumerable?  Seems like the overhead of doing so would simply be an extra virtual method call for each enumerable... with everything else happening, is that really a significant impact?